### PR TITLE
issue #28 - RegionInstanceGroupManager is forced to be re-created - fix.

### LINKED ISSUE #28: MIG module: RegionInstanceGroupManager is forced to be re-created on every provision
That causes resource already exist API error on every subsequent provision attempt: 

```
Error: Error creating RegionInstanceGroupManager: googleapi: Error 409: The resource 'projects/gl-ivankorniienko-seed-251912/regions/us-west1/instanceGroupManagers/multi-mig-lb-http-group1-mig' already exists, alreadyExists

  on .terraform/modules/mig1/terraform-google-modules-terraform-google-vm-cb3d6cc/modules/mig/main.tf line 32, in resource "google_compute_region_instance_group_manager" "mig":
  32: resource "google_compute_region_instance_group_manager" "mig" {



Error: Error creating RegionInstanceGroupManager: googleapi: Error 409: The resource 'projects/gl-ivankorniienko-seed-251912/regions/us-east1/instanceGroupManagers/multi-mig-lb-http-group2-mig' already exists, alreadyExists

  on .terraform/modules/mig2/terraform-google-modules-terraform-google-vm-cb3d6cc/modules/mig/main.tf line 32, in resource "google_compute_region_instance_group_manager" "mig":
  32: resource "google_compute_region_instance_group_manager" "mig" {

```

### PR DESCRIPTION
**Pull request by pksmall**
Fixes #28

Highlight:
```
~ distribution_policy_zones = [
          - "europe-west3-a",
          - "europe-west3-b",
          - "europe-west3-c",
        ] -> (known after apply) # forces replacement
```
ignore using **lifecycle**

### CODE REVIEW

**Comment by morgante** [comment]
@pksmall I'm not sure this is a sufficient fix, as it would prevent people from updating the desired zones of a MIG after creating it.

**Comment by pksmall** [comment]
@morgante Agree, but this will have to change:
```
  lifecycle {
    create_before_destroy = "true"
  }
```
to **false** 

I added to a test project *distribution_policy_zones* by groups and got same error.

**Comment by morgante** [comment]
I'm still unclear what is happening here though.

Without this, is Terraform claiming that the distribution zones have changed with every apply (even when they are untouched)?

**Comment by pksmall** [comment]
@morgante Exactly,
```
data "google_compute_zones" "available" {
  project = var.project_id
}
```
this definition is not constant for Terraform. Every apply Terraform overwrites a new.

**Comment by pksmall** [comment]
@morgante sure.

If you accept this fix, it would be nice to remove the unnecessary code from the module.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,14 @@ project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 
+## [1.1.1] - 2019-10-25
+
+### Fixed
+
+- `distribution_policy_zones` ignoring [#37]
```
Comment (morgante) [inline CHANGELOG.md]:
```suggestion
- Fix bug with `distribution_policy_zones` forcing permadiff. [#37]
```

Comment (morgante) [reply CHANGELOG.md]: Also, please don't put in the release numbers. You can just add this under "Unreleased"

File: README.md [hunk 2]
```diff
@@ -25,6 +25,11 @@ The following APIs must be enabled on your project:
 
 See also the [project_services](modules/project_services) module (optional).
 
+## Notes
+
+`distribution_policy_zones` cannot be changed during use. If you have changed them yourself or 
+used to a default value, then you'll have to force recreate a mig group yourself.
```
Comment (morgante) [inline README.md]:
```suggestion
used to have a default value, then you'll have to force recreate a MIG group yourself.
```

**Pull request merged**